### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by RonalAlexis2**
BACK-6 Ticket Solution

This is a hands-on object-oriented programming exercise, done in C# in ID Visual Studio.
The concepts of:

Class.
methods
attributes
Inheritance
extends
Implements
Polymorphism.
Interface
Objects as parameter
model class
encapsulation

Screenshot_1

### CODE REVIEW

**Review by MauroDecodificado — CHANGES_REQUESTED** [review]
Before starting code review, please check following comments

1. Your folder are the followed strcuture:
EjercicioUno-POO/EjerciciooUno/EjerciciooUno/{Here all your files}

There is not reason at this point to have such a structure
So please put all your code in the main root, it should look like this
EjercicioUno-POO/{Here all your files}

2. The PR title and description is not correct, so please update it nicely and put all the relevant information you consider the other people should know

3. Please add information to the read me in order to visualize what is this repo about some examples here https://github.com/crisgahur/OOPExercise

**Comment by RonalAlexis2** [comment]
Good morning, the indicated changes are made

File: ModeloPersona.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        // se crean las variables privadas para luego agregar Set y get
+
+        private string colorPelo;
+        private int edad;
+        private string nombre;
+
+
+        // se crean Get y Set
+
+        public string ColorPelo { get => colorPelo; set => colorPelo = value; }
```
Comment (MauroDecodificado) [inline ModeloPersona.cs]: This is not correct:

1. A variable can not be name with upper case at the beggining
2. You are creating two variable colorPelo and ColorPelo and that is not correct you have to create just one
3. To do the encapsulation on this you have to use the private variable for example 
For the variable colorPersona which is private you can create two method one to set the informatino and get the get the information from that variable, example:
```

        private string colorPelo;


        public String getColorPelo(){
            return colorPelo;
        }

        public setColorPelo(String colorPelo) {
             this.colorPelo = colorPelo;
        }
```

Comment (MauroDecodificado) [reply ModeloPersona.cs]: So delete those variables from line 21 to 23 and use just the attributes created at the beginning

File: Pedro.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EjerciciooUno
+{
+    internal class Pedro:Persona
```
Comment (MauroDecodificado) [inline Pedro.cs]: Please format code it must look like

  ```
  internal class Pedro : Persona
    {
```

File: Pedro.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
```
Comment (MauroDecodificado) [inline Pedro.cs]: Delete empty space just leave 1 space

File: Pedro.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    {
+        // se crea metodo para pedro camianr
+        public void caminar ()
```
Comment (MauroDecodificado) [inline Pedro.cs]: Delete the space between the name of the method and the parenthesis 

`public void caminar()`

File: Pedro.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        {
+            Console.WriteLine("pedro esta caminando");
+        }
+
+        //se crea metodo para pedro tocar piano
+        public void tocarPiano()
+
```
Comment (MauroDecodificado) [inline Pedro.cs]: Delete extra line

File: Pedro.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    {
+        // se crea metodo para pedro camianr
+        public void caminar ()
+        {
+            Console.WriteLine("pedro esta caminando");
+        }
+
+        //se crea metodo para pedro tocar piano
+        public void tocarPiano()
```
Comment (MauroDecodificado) [inline Pedro.cs]: Tocar piano following the exercise must recieve a parameter of type String and use it to print the message

File: Persona.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+            Console.WriteLine("gritando");
+        }
+
+        public void saltar()
+        {
+            Console.WriteLine("saltando");
+        }
+
+        // se crea variable global de tipo ModeloPersona
+
```
Comment (MauroDecodificado) [inline Persona.cs]: Delete empty space

File: Persona.cs [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+        // se crea metodo llenar info que recibe datos1 de tipo ModeloPersona y guardamos ese dato1 en la variable global
+        public void llenarInfo (ModeloPersona datos1)
+
+        {
+            variableGlobal = datos1;
+        }
+
+
```
Comment (MauroDecodificado) [inline Persona.cs]: Delete all those space below

File: Persona.cs [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+        }
+
+        // se crea variable global de tipo ModeloPersona
+
+
+         public ModeloPersona variableGlobal;
+
+
+        // se crea metodo llenar info que recibe datos1 de tipo ModeloPersona y guardamos ese dato1 en la variable global
+        public void llenarInfo (ModeloPersona datos1)
```
Comment (MauroDecodificado) [inline Persona.cs]: Avoid using attributes called datos1, datos2, etc

Please rename it to something more meaningful, for example datosPersona

File: Pepita.cs [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+using System.Threading.Tasks;
+
+namespace EjerciciooUno
+{
+    internal class Pepita:Persona
+    {
+        // se crea metodo para pepita tocar flauta
+        public void tocarFlauta()
+        {
+
```
Comment (MauroDecodificado) [inline Pepita.cs]: Please delete this emtpy line

File: Pepita.cs [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace EjerciciooUno
+{
+    internal class Pepita:Persona
+    {
+        // se crea metodo para pepita tocar flauta
+        public void tocarFlauta()
```
Comment (MauroDecodificado) [inline Pepita.cs]: tocar flauta must receive a param as String with the song

File: Pepita.cs [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+        {
+
+            Console.WriteLine("pepita toca flauta");
+        }
+
+        // se crea metodo para pepita caminar
+        public void caminar()
```
Comment (MauroDecodificado) [inline Pepita.cs]: The exercise do not ask to override caminar from the parent persona

File: Pedro.cs [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+        
+        {
+
+            Console.WriteLine("pedro toca piano");
+        }
+
```
Comment (MauroDecodificado) [inline Pedro.cs]: Please delete empty lines below

File: Program.cs [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+namespace EjerciciooUno
+{
+    internal class Program
+    {
+        static void Main(string[] args)
+        {
+            
+            // se instancia pedro             
+            Pedro opedro = new Pedro ();
```
Comment (MauroCaro) [inline Program.cs:15]: calle this variable pedro instead of `opedro `

File: Program.cs [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+
+            // se crea modelo persona de pedro
+            ModeloPersona modelopedro = new ModeloPersona ("Blanco",45,"pedro perez");
+
+            //se llena info de pedro
+            opedro.llenarInfo (modelopedro);
```
Comment (MauroDecodificado) [inline Program.cs]: Delete space between method and parameters 


 `opedro.llenarInfo(modelopedro);`

File: Program.cs [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+
+            // Se llama tocar piano           
+            opedro.tocarPiano();
```
Comment (MauroDecodificado) [inline Program.cs]: Since tocarPiano must receive a string you have to send the song here

File: Program.cs [hunk 17]
```diff
@@ -0,0 +1,63 @@
...
+
+            //se llama caminar
+            opedro.caminar();
+
+            // se muestra informacion de pedro almacenada
+
+            Console.Write("pedro tiene el pelo de color " + opedro.variableGlobal.ColorPelo);
```
Comment (MauroDecodificado) [inline Program.cs]: You can wrap these three console into one by doing

` Console.Write("pedro tiene el pelo de color " + opedro.variableGlobal.ColorPelo + " su edad es "+ opedro.variableGlobal.Edad +" y su nombre es " + opedro.variableGlobal.Nombre );`

File: Program.cs [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+            ModeloPersona modelopepita = new ModeloPersona("Rubio", 25, "pepita perez");
+
+            //se llena info de pepita
+            opepita.llenarInfo(modelopepita);
+
+            // se llama caminar en pepita
+            opepita.caminar();
+
+            // se llama tocar flauta
+            opepita.tocarFlauta();
```
Comment (MauroDecodificado) [inline Program.cs]: Since tocarFlauta must receive a string you have to send the song here

File: Program.cs [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+
```
Comment (MauroDecodificado) [inline Program.cs]: Delete empty spaces

File: Program.cs [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+
+
+
+
+            Console.ReadKey();
+
```
Comment (MauroDecodificado) [inline Program.cs]: Delete empty spaces

File: README.md [hunk 21]
```diff
@@ -1,2 +1,35 @@
-# EjercicioUno-POO
-Ejercicio para desarrollar habilidades y conocimientos en programacion orientada a objetos
+# POOExercise ONE
```
Comment (MauroDecodificado) [inline README.md:1]: Niceee 👏

**Review by MauroDecodificado — CHANGES_REQUESTED** [review]
Please check comments also for inden code use contro+K+D

File: Pedro.cs (hunk 4 above)
Comment (RonalAlexis2) [reply Pedro.cs]: has been corrected

File: Pedro.cs (hunk 3 above)
Comment (RonalAlexis2) [reply Pedro.cs]: has been corrected

File: Pedro.cs (hunk 2 above)
Comment (RonalAlexis2) [reply Pedro.cs]: has been corrected

File: Pedro.cs (hunk 5 above)
Comment (RonalAlexis2) [reply Pedro.cs]: has been corrected

File: Pedro.cs (hunk 6 above)
Comment (RonalAlexis2) [reply Pedro.cs]: has been corrected

File: Persona.cs (hunk 7 above)
Comment (RonalAlexis2) [reply Persona.cs]: has been corrected

File: Persona.cs (hunk 8 above)
Comment (RonalAlexis2) [reply Persona.cs]: has been corrected

File: Persona.cs (hunk 9 above)
Comment (RonalAlexis2) [reply Persona.cs]: has been corrected

File: Pepita.cs (hunk 10 above)
Comment (RonalAlexis2) [reply Pepita.cs]: has been corrected

File: Pepita.cs (hunk 11 above)
Comment (RonalAlexis2) [reply Pepita.cs]: has been corrected

File: Pedro.cs (hunk 13 above)
Comment (RonalAlexis2) [reply Pedro.cs]: has been corrected

File: Program.cs (hunk 15 above)
Comment (RonalAlexis2) [reply Program.cs]: has been corrected

File: Program.cs (hunk 16 above)
Comment (RonalAlexis2) [reply Program.cs]: has been corrected

File: Program.cs (hunk 17 above)
Comment (RonalAlexis2) [reply Program.cs]: has been corrected

File: Program.cs (hunk 18 above)
Comment (RonalAlexis2) [reply Program.cs]: has been corrected

File: Program.cs (hunk 19 above)
Comment (RonalAlexis2) [reply Program.cs]: has been corrected

File: Program.cs (hunk 20 above)
Comment (RonalAlexis2) [reply Program.cs]: has been corrected

File: ModeloPersona.cs (hunk 1 above)
Comment (RonalAlexis2) [reply ModeloPersona.cs]: has been corrected

File: Pepita.cs (hunk 12 above)
Comment (RonalAlexis2) [reply Pepita.cs]: has been corrected

File: Persona.cs [hunk 22]
```diff
@@ -27,25 +26,15 @@ public void saltar()
...
-
         // se crea metodo llenar info que recibe datos1 de tipo ModeloPersona y guardamos ese dato1 en la variable global
-        public void llenarInfo (ModeloPersona datos1)
-
+        public void llenarInfo(ModeloPersona datosPersona1)
         {
-            variableGlobal = datos1;
-        }
-
+            variableGlobal = datosPersona1;
```
Comment (MauroDecodificado) [inline Persona.cs]: Please rename this to be `datosPersona`

File: Persona.cs [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+
+        // se crea variable global de tipo ModeloPersona
+         public ModeloPersona variableGlobal;
+
+        // se crea metodo llenar info que recibe datos1 de tipo ModeloPersona y guardamos ese dato1 en la variable global
+        public void llenarInfo(ModeloPersona datosPersona1)
+        {
+            variableGlobal = datosPersona1;
+        }              
+        // se crea metodo para pepita caminar
```
Comment (MauroDecodificado) [inline Persona.cs]: I don't think this comment belong to this class, please check

**Review by MauroDecodificado — CHANGES_REQUESTED** [review]
Just couple comments, please check

File: Persona.cs (hunk 22 above)
Comment (RonalAlexis2) [reply Persona.cs]: has been corrected

File: Persona.cs (hunk 23 above)
Comment (RonalAlexis2) [reply Persona.cs]: has been corrected

Comment (RonalAlexis2) [reply Persona.cs]: Corrections are made to the indicated comments, I stay tuned to perform the merge

**Review by MauroDecodificado — APPROVED** [review]
PR is good, awesome Job 👏  You can merge it now

**Pull request merged**